backend/de.metas.contracts/src/main/java/de/metas/contracts/finalinvoice/invoicecandidate/FlatrateTermModular_Handler.java

```diff
@@ -0,0 +1,312 @@
+
+		return invoiceCandidatesAll.build();
+	}
+
+	@Override
+	@NonNull
+	public ImmutableList<Object> getRecordsToLock(@NonNull final I_C_Flatrate_Term term)
+	{
+		final ImmutableList.Builder<Object> recordsToLock = ImmutableList.builder();
+
+		recordsToLock.add(term);
+
+		final ModularContractLogQuery query = ModularContractLogQuery.builder()
+				.flatrateTermId(FlatrateTermId.ofRepoId(term.getC_Flatrate_Term_ID()))
+				.processed(false)
+				.billable(true)
+				.build();
+
+		recordsToLock.addAll(modularContractLogDAO.getModularContractLogRecords(query)
+									 .stream()
+									 .collect(ImmutableList.toImmutableList()));
+
+		return recordsToLock.build();
+	}
+
+	@Override
+	public Instant calculateDateOrdered(@NonNull final I_C_Invoice_Candidate invoiceCandidateRecord)
+	{
+		return SystemTime.asInstant();
+	}
+
+	@Override
+	@NonNull
+	public ProductId getProductId(@NonNull final I_C_Flatrate_Term term, @NonNull final I_C_Invoice_Candidate ic)
+	{
+		return Optional.ofNullable(ProductId.ofRepoIdOrNull(ic.getM_Product_ID()))
+				.orElseThrow(() -> new AdempiereException("Product cannot be missing at this point !")
+						.appendParametersToMessage()
+						.setParameter("C_Flatrate_Term_ID", term.getC_Flatrate_Term_ID()));
+	}
+
+	@NonNull
+	private I_C_Invoice_Candidate createCandidateFor(
+			@NonNull final I_C_Flatrate_Term modularContract,
+			@NonNull final ModuleConfig moduleConfig,
+			@NonNull final YearAndCalendarId yearAndCalendarId,
+			@Nullable final PricingSystemId pricingSystemId)
+	{
+		final I_C_Invoice_Candidate invoiceCandidate = InterfaceWrapperHelper.newInstance(I_C_Invoice_Candidate.class);
+
+		invoiceCandidate.setAD_Table_ID(InterfaceWrapperHelper.getTableId(I_C_Flatrate_Term.class));
+		invoiceCandidate.setRecord_ID(modularContract.getC_Flatrate_Term_ID());
+		invoiceCandidate.setAD_Org_ID(modularContract.getAD_Org_ID());
+		invoiceCandidate.setC_Flatrate_Term_ID(modularContract.getC_Flatrate_Term_ID());
+		invoiceCandidate.setC_Async_Batch_ID(modularContract.getC_Async_Batch_ID());
+
+		InvoiceCandidateLocationAdapterFactory
+				.billLocationAdapter(invoiceCandidate)
+				.setFrom(ContractLocationHelper.extractBillLocation(modularContract));
+
+		invoiceCandidate.setIsTaxIncluded(modularContract.isTaxIncluded());
+
+		//
+		invoiceCandidate.setLine(moduleConfig.getSeqNo().toInt());
+		invoiceCandidate.setM_Product_ID(moduleConfig.getProductId().getRepoId());
+		invoiceCandidate.setInvoicingGroup(moduleConfig.getInvoicingGroup());
+		invoiceCandidate.setProductName(moduleConfig.getName());
+		//
+		invoiceCandidate.setHarvesting_Year_ID(yearAndCalendarId.yearId().getRepoId());
+		invoiceCandidate.setC_Harvesting_Calendar_ID(yearAndCalendarId.calendarId().getRepoId());
+		//
+		invoiceCandidate.setM_PricingSystem_ID(PricingSystemId.toRepoId(pricingSystemId));
+
+		return invoiceCandidate;
+	}
+
+	@NonNull
+	private TaxId getTaxId(
```

Comment:
looks like this is not used

Reply:
also why we would use that instead of  de.metas.invoicecandidate.api.impl.InvoiceCandBL#getTaxEffective?